### DISCUSSION TITLE
Integrating sqlite3

### DISCUSSION THREAD

**Discussion by sketchylizard** [cpm-cmake/CPM.cmake#351] [General]
Has anyone tried integrating sqlite3? There is a GitHub mirror for the sources but they don't use CMake. There are a few GitHub repos that mirror the sqlite-amalgamation and include CMake files though. I've added that using:

CPMAddPackage(
    NAME sqlite3
    GIT_REPOSITORY https://github.com/azadkuh/sqlite-amalgamation.git
    GIT_TAG v3.8.10.2)

But it's not building and I can't seem to use it in any calls to target_link_libraries.

**Comment by sketchylizard** [comment]
I'm now getting it to build sqlite, but calling target_link_libraries(MyProject PUBLIC SQLite3) does not add any directories to my include path.

**Comment by andijcr** [comment]
what i did:
```
CPMAddPackage(
    NAME sqlite3
    VERSION 3.37.2
    DOWNLOAD_ONLY TRUE
    URL "https://www.sqlite.org/2022/sqlite-amalgamation-3370200.zip"
    )
if(sqlite3_ADDED)
    find_package(Threads REQUIRED)
    add_library(sqlite3 STATIC ${sqlite3_SOURCE_DIR}/sqlite3.c ${sqlite3_SOURCE_DIR}/sqlite3.h)
    target_include_directories(sqlite3 PUBLIC ${sqlite3_SOURCE_DIR})
    target_link_libraries(sqlite3 PUBLIC Threads::Threads)
    target_compile_definitions(sqlite3 PRIVATE SQLITE_OMIT_LOAD_EXTENSION)
endif()
```
this will add a static library `sqlite3`, with the `SQLITE_OMIT_LOAD_EXTENSION` configuration
